witness/src/Dscp/Witness/Web/Logic.hs

```diff
@@ -94,3 +95,24 @@ getTransactionInfo =
     runSdMRead . getTxMaybe >=>
     nothingToThrow TransactionNotFound >=>
     return . toTxInfo
+
+getHashType :: WitnessWorkMode ctx m => Text -> m HashIs
+getHashType hash = fmap (fromMaybe HashIsUnknown) . runMaybeT . asum $
+    [isBlock, isAddress, isTx] <*> [hash]
+  where
+    is hashType decode getMaybe =
+        MaybeT . return . rightToMaybe . decode >=>
+        MaybeT . getMaybe >=>
+        MaybeT . return . Just . const hashType
+    isBlock = is
+        HashIsBlock
+        (fromHex @HeaderHash)
+        (runSdMRead . getBlockMaybe)
+    isAddress = is
+        HashIsAddress
+        addrFromText
+        getAccountMaybe
+    isTx = is
+        HashIsTx
+        fromHex
+        (runSdMRead . getTxMaybe . GTxId)
```

Comment:
Is `getMaybe` part really required?

Reply:
Otherwise, I can't distinguish between block and tx hashes, they are exactly same.

Reply:
Right. I think you should leave a comment about purpose you mentioned, and also mention in swagger that entity should exist.

Reply:
It's ambiguous whether entity is unknown or type of entity is unknown. I personally though the former at first.

Reply:
`description: Get type of entity identified by given hash`
If we don't know any entity with this hash, we don't the type of it, and vice versa.
This does not hold for addresses, as their format is less ambiguous, but it's irrelevant for the purposes of this API.